### PR TITLE
Updated PSR Container to v2

### PR DESCRIPTION
**Pull request by route33**
All tests are passing

PHPUnit 9.5.10 by Sebastian Bergmann and contributors.

...............................................................  63 / 233 ( 27%)
............................................................... 126 / 233 ( 54%)
............................................................... 189 / 233 ( 81%)
............................................                    233 / 233 (100%)

Time: 00:01.703, Memory: 10.00 MB

OK (233 tests, 254 assertions)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,8 +15,8 @@
         "source": "https://github.com/woohoolabs/zen"
     },
     "require": {
-        "php": "^8.0.0",
-        "psr/container": "^1.0.0",
+        "php": ">=8.0",
+        "psr/container": "^1.0 || ^2.0",
```
Comment (kocsismate) [inline composer.json:19]: The current implementation is not compatible with PSR/Container 1.0 anymore. Only 1.1+ should be supported.

File: .gitignore [hunk 2]
```diff
@@ -6,3 +6,4 @@ examples/Container.php
 examples/Definitions/
 examples/preload.php
 tests/Definitions/
+.idea
```
Comment (kocsismate) [inline .gitignore:9]: system-specific ignores should go to your global gitignore (https://gist.github.com/subfuzion/db7f57fff2fb6998a16c)

File: composer.json [hunk 3]
```diff
@@ -15,8 +15,8 @@
         "source": "https://github.com/woohoolabs/zen"
     },
     "require": {
-        "php": "^8.0.0",
-        "psr/container": "^1.0.0",
+        "php": ">=8.0",
```
Comment (kocsismate) [inline composer.json:18]: I wouldn't like to loosen the PHP compatibility requirement this much. If time comes (there are at least 2 more years), and this version of Zen is still compatible with PHP 9, then I'll release a patch version adding supporting for it.

Comment (route33) [reply composer.json]: True, I had nightmares last night about rejecting the PR because of this change.

**Comment by kocsismate** [comment]
Thank you very much for the help! It's going to be added to the 4.0 release.

**Pull request merged**